features/networking/ipv6_dualstack.feature

```diff
@@ -0,0 +1,78 @@
+Feature: ipv6 dual stack cluster test scenarios
+
+  # @author zzhao@redhat.com
+  # @case_id OCP-40581
+  @admin
+  Scenario: Perject should be isolation when using multitenant policy for ipv6 dual stack
+    # create project and pods
+    Given I have a project
+    And evaluation of `project.name` is stored in the :proj1 clipboard
+    Given I obtain test data file "networking/list_for_pods_ipv6.json"
+    When I run the :create client command with:
+      | f | list_for_pods_ipv6.json |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(0).ip_v4` is stored in the :p1pod1ipv4 clipboard
+    And evaluation of `pod(0).ip_v6` is stored in the :p1pod1ipv6 clipboard
+    Given I use the "test-service" service
+    And evaluation of `service.ip_v4` is stored in the :service_ipv4 clipboard
+    And evaluation of `service.ip_v6` is stored in the :service_ipv6 clipboard
+
+    # create another project and pods
+    Given I create a new project
+    And evaluation of `project.name` is stored in the :proj2 clipboard
+    Given I obtain test data file "networking/list_for_pods_ipv6.json"
+    When I run the :create client command with:
+      | f | list_for_pods_ipv6.json |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(1).name` is stored in the :p2pod1 clipboard
+
+    When I execute on the "<%= cb.p2pod1 %>" pod:
+      | curl | --connect-timeout | 5 | <%= cb.p1pod1ipv4 %>:8080 |
+    Then the step should succeed
+    And the output should contain "Hello"
+    When I execute on the "<%= cb.p2pod1 %>" pod:
+      | curl | --connect-timeout | 5 | [<%= cb.p1pod1ipv6 %>]:8080 |
+    Then the step should succeed
+    And the output should contain "Hello"
+
+    #access pod by service ipv4 and ipv6 address
+    When I execute on the "<%= cb.p2pod1 %>" pod:
+      | curl | --connect-timeout | 5 | <%= cb.service_ipv4 %>:27017 |
```

Comment:
I guess what I am suggesting for ports is that we get in the habit of using `service.ports[0]["port"]` to automatically use the first port from the service spec.

Maybe we need a `.port_index` method.

```ruby
    def port_index(user: nil, num:, cached: true, quiet: false)
      rr = raw_resource(user: user, cached: cached, quiet: quiet)
      return rr.dig('spec', 'ports')[num]["port"]
    end
```
```ruby
| curl | --connect-timeout | 5 | <%= cb.service_ipv4 %>:<%= service.port_index(0) %> |
```

Reply:
update the ip_v6_url:
```
    def ip_v6_url(user: nil, cached: true, quiet: false)
      ipv6 = self.ip_v6(user: user, cached: cached, quiet: quiet)
      port = self.ports(user: user, cached: true, quiet: quiet)[0]["port"]
      "[#{ipv6}]:#{port}"
    end
```